docs/manual/select_and_filter_data.rst

```diff
@@ -0,0 +1,1173 @@
+   >>>
+   >>> source = {"dates" : [date(2000, 1, 5), date(2010, 11, 23), date(2020, 2, 29), None], 
+   ...           "integers" : range(1, 5), 
+   ...           "floats" : [10.0, 11.5, 12.3, -13], 
+   ...           "strings" : ['A', 'B', None, 'D']
+   ...           }
+   >>> DT = dt.Frame(source)
+   >>> DT
+      | dates       integers   floats  strings
+      | date32         int32  float64  str32  
+   -- + ----------  --------  -------  -------
+    0 | 2000-01-05         1     10    A      
+    1 | 2010-11-23         2     11.5  B      
+    2 | 2020-02-29         3     12.3  NA     
+    3 | NA                 4    -13    D      
+   [4 rows x 4 columns]
+
+Column selection is possible via a number of options:
+
+By column name
+^^^^^^^^^^^^^^
+::
+
+   >>> DT[:, 'dates'] 
+      | dates     
+      | date32    
+   -- + ----------
+    0 | 2000-01-05
+    1 | 2010-11-23
+    2 | 2020-02-29
+    3 | NA        
+   [4 rows x 1 column]
+
+.. note:: When returning all rows, the ``i`` section can be denoted by ``:``, or ``None``, or ``...``
+
+By position
+^^^^^^^^^^^
+::
+
+   >>> DT[..., 2]
+      |  floats
+      | float64
+   -- + -------
+    0 |    10  
+    1 |    11.5
+    2 |    12.3
+    3 |   -13  
+   [4 rows x 1 column]
+
+With position, you can select with a negative number - the column will be selected from the end; this is similar to indexing a python list::
+
+   >>> DT[None, -2]    
+      |  floats
+      | float64
+   -- + -------
+    0 |    10  
+    1 |    11.5
+    2 |    12.3
+    3 |   -13  
+   [4 rows x 1 column]
+
+.. note:: For a single column, it is possible to skip the ``:`` in the `i` section and pass the column name or position only
+
+   >>> DT['dates']
+      | dates     
+      | date32    
+   -- + ----------
+    0 | 2000-01-05
+    1 | 2010-11-23
+    2 | 2020-02-29
+    3 | NA        
+   [4 rows x 1 column]
+   ... DT[0] 
+      | dates     
+      | date32    
+   -- + ----------
+    0 | 2000-01-05
+    1 | 2010-11-23
+    2 | 2020-02-29
+    3 | NA        
+   [4 rows x 1 column]
+
+.. note:: When selecting via column name or position, an error is returned if the name or position does not exist
+
+   >>> DT[:, 5]
+   >>> ---------------------------------------------------------------------------
+   >>> ValueError                                Traceback (most recent call last)
+   >>> <ipython-input-63-c0dafcf7de2e> in <module>
+   >>> ----> 1 DT[:, 5]
+   >>>
+   >>> ValueError: Column index 5 is invalid for a Frame with 4 columns
+
+   >>> DT[:, 'categoricals']
+   >>> ---------------------------------------------------------------------------
+   >>> KeyError                                  Traceback (most recent call last)
+   >>> <ipython-input-64-0b4834f1b510> in <module>
+   >>> ----> 1 DT[:, 'categoricals']
+   >>>
+   >>> KeyError: Column categoricals does not exist in the Frame
+
+By Data Type
+^^^^^^^^^^^^
+Column selection is possible by using python's `built-in types`_:
```

Comment:
I guess we should say here that not all the python's built-in types will work, but only those that have the corresponding datatable ltypes.